plugins/module_utils/network/grpc/grpc.py

```diff
@@ -0,0 +1,156 @@
+# (c) 2022 Red Hat, Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import json
+import re
+
+from ansible.module_utils._text import to_text
+from ansible.module_utils.connection import Connection
+
+
+def get_connection(module):
+    if hasattr(module, "_grpc_connection"):
+        return module._grpc_connection
+
+    capabilities = get_capabilities(module)
+    network_api = capabilities.get("network_api")
+    if network_api == "ansible.netcommon.grpc":
+        module._grpc_connection = Connection(module._socket_path)
+    else:
+        module.fail_json(msg="Invalid connection type %s" % network_api)
```

Comment:
Is this still needed if we're already checking this in the action plugin?

Reply:
action plugin will be invoked only for `grpc_config` module as the name of the action plugin file is `grpc_config.py` 
The name of the action plugin file can be renamed to `grpc.py` and redirection for `grpc_config` and `grpc_get` modules can be added to the action plugin in `runtime.yml` file.

Reply:
Removing grpc.py for now

Reply:
Done